docs/DeveloperGuide.md

```diff
@@ -1,34 +1,920 @@
+
+The `WorkspaceCommand` is instantiated by `WorkspaceParser`, and it requires 2 parameters: 
+*   `commandOption` (mandatory).
+*   `workspaceName` (mandatory unless option `-l` is specified).
+
+<br/>
+
+Given below is an example usage scenario showing how the command behaves at each step when the user tries 
+to **create new** `Workspace`:
+
+**Step 1:** User launches the application for the first time. The `User` will be initialized with an initial `Workspace` named `Default`, and the `activeWorkspace` pointing to it and `workspaceList` `ArrayList` containing it.
+
+![Workspace Command Initial State Diagram](images/WorkspaceCommand-Initial-State.png) <br/>
+*Figure 14: Workspace Command Initial State*
+
+<br/>
+
+**Step 2:** User enters the command `workspace -n Netflix Animation Studio`, the input will be processed and parsed by `Parser.java` and then further parsed by `WorkspaceParser.java`.
+
+**Step 3:** Upon completion of parsing and input validation, `WorkspaceParser.java` will create a `WorkspaceCommand` object with the extracted `commandOption` and `workspaceName` parameter and return it to `Main`.
+
+**Step 4:** `Main` calls `WorkspaceCommand#execute()` and it checks the `commandOption` before running `WorkspaceCommand#createWorkspace()` accordingly.
+
+**Step 5:** `WorkspaceCommand` firstly calls `User#addWorkspace()` to add a new `Workspace` to `User`, then makes an empty `ArrayList` of `Watchlist` using `User#setWatchlistList` for the `User`.
+Finally, it uses `storageManager#saveWorkspace()` to save the `Workspace` to disk.
+
+![Workspace Command After Creation Diagram](images/WorkspaceCommand-After-Create.png) <br/>
+*Figure 15: Workspace Command After New Workspace Creation*
+
+<br/>
+
+**Step 6:** If successful, `WorkspaceCommand` returns the successfully created workspace message to `Main`.
+
+<br/>
+
+Likewise, the operations to switch, list, and delete follows a similar execution process. 
+The following diagrams will continue **from step 6**, and will illustrate the changes to the `Workspace` `ArrayList`.
+
+**Step 7:** User keys in `workspace -s Netflix Animation Studio` to switch active workspace.
+
+![Workspace Command After Switch Diagram](images/WorkspaceCommand-After-Switch.png) <br/>
+*Figure 16: Workspace Command After Workspace Switch*
+
+<br/>
+
+**Step 8:** User keys in `workspace -d Default` to delete the workspace named `Default`.
+
+![Workspace Command After Switch Diagram](images/WorkspaceCommand-After-Delete.png) <br/>
+*Figure 17: Workspace Command After New Workspace Delete*
+
+<br/>
+
+The following sequence diagram illustrates how `Workspace` creation in the example above works:
+
+> :memo: The other options (`-s`, `-l`, `-d`) follows a similar process, only the list and switch option does not interact with `StorageManager` and `Watchlist`.
+
+![Workspace Command Sequence Diagram](images/WorkspaceCommand-Sequence-Diagram.png) <br/>
```

Comment:
![image](https://user-images.githubusercontent.com/57039314/97526516-4a9d6900-19e4-11eb-86bb-9c59faa88d2b.png)

Just to check, the box seems to be missing the label (eg. alt, opt)

Reply:
Hello, may I know which box you're referring to?